core/iwasm/compilation/debug/dwarf_extractor.cpp

```diff
@@ -315,6 +315,20 @@ lldb_function_to_function_dbi(const AOTCompContext *comp_ctx,
         }
     }
 
+    auto compile_unit = sc.GetCompileUnit();
+    auto file_spec = compile_unit.GetFileSpec();
+    const char *file_name = file_spec.GetFilename();
+    const char *dir_name = file_spec.GetDirectory();
+    LLVMMetadataRef file_info = NULL;
+    if (file_name || dir_name) {
+        file_info = LLVMDIBuilderCreateFile(comp_ctx->debug_builder, file_name,
+                                            strlen(file_name), dir_name,
```

Comment:
potential UB if `file_name` or `dir_name` is null? 

I'm not aware of the context but I think I should be either 

1. `if (file_name && dir_name)`
2. using `strnlen_s` 

https://en.cppreference.com/w/c/string/byte/strlen

Reply:
Not sure whether file_name and dir_name can be one is NULL and the other isn't, if yes, had better be somewhat like below?
```
        file_info = LLVMDIBuilderCreateFile(comp_ctx->debug_builder,
                                            file_name,
                                            file_name ? strlen(file_name) : 0,
                                            dir_name,
                                            dir_name ? strlen(dir_name) : 0);
```
If not, had better use `if (file_name && dir_name)`.

Reply:
probably. this line is a copy-and-paste from other function in the same file.

Reply:
i made them safer just in case.